app/main/forms.py

```diff
@@ -2591,18 +2592,142 @@ class PostcodeForm(StripWhitespaceForm):
+    )
+
+    def pre_validate(self, form):
+        return self.postcode.validate(form)
+
+    def post_validate(self):
+        return not self.errors and not self.postcode.errors and not self.radius.errors
+
+
+class DecimalCoordinatesForm(StripWhitespaceForm):
+    """
+    This form contains the input fields for creation of areas using decimal coordinates i.e. Latitude & longitude.
+    For the first coordinate, latitude, to be valid the value entered must be numeric and between -90 and 90,
+    hence the NumberRange validator.
+    For the second coordinate, longitude, to be valid the value again must be number and between -180 and 180.
+    There is additional validation post-submission of the form that checks whether the coordinate lies within
+    either the UK or the test polygons in Finland, and if both are false then an error is appended to the form.
+    """
+
+    first_coordinate = GovukDecimalField(
+        "Latitude",
+        validators=[
+            InputRequired("Enter a valid latitude"),
+            NumberRange(min=-90.0099, max=90.001, message="Enter a latitude between -90 and 90"),
+            Only6DecimalPlaces(),
+        ],
+        param_extensions={
+            "classes": "govuk-input govuk-input--width-6",
+            "attributes": {"pattern": "^-?\\d+(\\.\\d+)?$"},
+        },
+    )
+    second_coordinate = GovukDecimalField(
+        "Longitude",
+        validators=[
+            InputRequired("Enter a valid longitude"),
+            Only6DecimalPlaces(),
+            NumberRange(min=-180.0099, max=180.001, message="Enter a longitude between -180 and 180"),
+        ],
+        param_extensions={
+            "classes": "govuk-input govuk-input--width-6",
+            "attributes": {"pattern": "^-?\\d+(\\.\\d+)?$"},
+        },
+    )
+    radius = GovukDecimalField(
+        "Radius",
+        param_extensions={
+            "classes": "govuk-input govuk-input--width-5",
+            "suffix": {"text": "km"},
+            "attributes": {"pattern": "^-?\\d+(\\.\\d+)?$"},
+        },
+        validators=[
+            NumberRange(min=0.099, max=38.001, message="Enter a radius between 0.1km and 38.0km"),
+            DataRequired(message="Enter a radius"),
+            Only2DecimalPlaces(),
+        ],
+    )
+
+    def post_validate(self):
+        return not self.errors
+
+    def pre_validate(self, form):
+        self.first_coordinate.validate(form)
+        self.second_coordinate.validate(form)
+        return self.first_coordinate.validate(form) and self.second_coordinate.validate(form)
+
+
+class CartesianCoordinatesForm(StripWhitespaceForm):
+    """
+    This form contains the input fields for creation of areas using cartesian coordinates i.e. Eastings & northings.
+    There is additional validation post-submission of the form that checks whether the coordinate lies within
+    either the UK or the test polygons in Finland, and if both are false then an error is appended to the form.
+    """
```

Comment:
We're unlikely to use this for anything other than Eastings/Northings I'd be tempted just to call this EastingNorthingCoordinatesForm.  That way we also don't have to manually mentally translate between cartesian and eastings/northings in the rest of the codebase.